src/components/AddComment.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from "react";
+import { postComment } from "./utils/utils";
+import "../../src/App.css";
+
+const AddComment = ({ article_id, setComments }) => {
+  const [comment, setComment] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isCommentPosted, setIsCommentPosted] = useState(false);
+  const [handleSubmitError, setHandleSubmitError] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!isCommentPosted && !isSubmitting) {
+      setIsSubmitting(true);
+      setHandleSubmitError(false);
+      postComment(article_id, comment)
+        .then((newComment) => {
+          setIsCommentPosted(true);
+          setComment("");
+          setComments((prevComments) => [newComment, ...prevComments]);
+        })
+        .catch((err) => {
+          setHandleSubmitError(true);
+        })
+        .finally(() => {
+          setIsSubmitting(false);
+          setIsCommentPosted(false);
+        });
+    }
+  };
+  return (
+    <form onSubmit={handleSubmit} className="comment-form">
+      <h2>Post a comment:</h2>
+      <textarea
```

Comment:
The 'resizing' of the textarea could be disabled, but it's totally up to you. This is just a reminder. 🙂